website/mailchimp_utils.py

```diff
@@ -67,18 +67,20 @@ def subscribe_mailchimp(list_name, user_id):
 @queued_task
 @app.task
 @transaction()
-def unsubscribe_mailchimp(list_name, user_id):
+def unsubscribe_mailchimp(list_name, user_id, username=None):
     """Unsubscribe a user from a mailchimp mailing list given its name.
 
     :param str list_name: mailchimp mailing list name
-    :param str username: current user's email
+    :param str user_id: current user's id
+    :param str username: current user's email (required for merged users)
 
     :raises: ListNotSubscribed if user not already subscribed
     """
     user = User.load(user_id)
+    username = user.username if user else username
     m = get_mailchimp_api()
     list_id = get_list_id_from_name(list_name=list_name)
-    m.lists.unsubscribe(id=list_id, email={'email': user.username})
+    m.lists.unsubscribe(id=list_id, email={'email': username})
```

Comment:
Since `username` isn't used elsewhere here, I'd prefer to not have defined it:

```python
m.lists.unsubscribe(id=list_id,
                    email={'email': user.username if user else username})
````

While that saves the Python intrepreter from having ot allocate space for a variable, that's insignificant. The real benefit I see would be that the definition of the variable is two lines away from its usage. If it's used for something else in the future, the purpose of that conditional might be lost, and the conditions could be changed without understanding the reason it is the way it is.

This isn't something I think needs to be changed, just a general preference on my part.